lambdas/pkgpush/src/t4_lambda_pkgpush/__init__.py

```diff
@@ -475,176 +474,213 @@ def get_pkg(src_registry, data):
 
+    registry_url = f"s3://{params.bucket}"
     try:
-        package_registry = get_registry(registry)
+        package_registry = get_registry(registry_url)
 
-        meta = data.get('meta')
-        message = data.get('message')
-        quilt3.util.validate_package_name(handle)
+        quilt3.util.validate_package_name(params.name)
         pkg = quilt3.Package()
-        if meta is not None:
-            pkg.set_meta(meta)
+        if params.user_meta is not None:
+            pkg.set_meta(params.user_meta)
 
         size_to_hash = 0
         files_to_hash = 0
-        for entry in map(get_schema_validator(PACKAGE_CREATE_ENTRY_SCHEMA), json_iterator):
+        for entry in map(PackageConstructEntry.parse_raw, req_file):
             try:
-                physical_key = PhysicalKey.from_url(entry['physical_key'])
+                physical_key = PhysicalKey.from_url(entry.physical_key)
             except ValueError:
                 raise PkgpushException(
                     "InvalidS3PhysicalKey",
-                    {"physical_key": entry['physical_key']},
+                    {"physical_key": entry.physical_key},
                 )
             if physical_key.is_local():
                 raise PkgpushException(
                     "InvalidLocalPhysicalKey",
                     {"physical_key": str(physical_key)},
                 )
-            logical_key = entry['logical_key']
-
-            hash_ = entry.get('hash')
-            obj_size = entry.get('size')
-            # `meta` is the full metadata dict for entry that includes
-            # optional `user_meta` property,
-            # see PackageEntry._meta vs PackageEntry.meta.
-            meta = entry.get('meta')
 
-            if hash_ and obj_size is not None:
+            if entry.hash and entry.size is not None:
                 pkg.set(
-                    logical_key,
+                    entry.logical_key,
                     quilt3.packages.PackageEntry(
                         physical_key,
-                        None if obj_size is None else int(obj_size),
-                        {'type': 'SHA256', 'value': hash_},
-                        meta,
-                    )
+                        entry.size,
+                        entry.hash.dict(),
+                        entry.meta,
+                    ),
                 )
             else:
-                pkg.set(logical_key, str(physical_key))
-                pkg[logical_key]._meta = meta or {}
+                pkg.set(entry.logical_key, str(physical_key))
+                pkg_entry = pkg[entry.logical_key]
+                assert isinstance(pkg_entry, quilt3.packages.PackageEntry)
+                pkg_entry._meta = entry.meta or {}
 
-                size_to_hash += pkg[logical_key].size
+                assert isinstance(pkg_entry.size, int)
+                size_to_hash += pkg_entry.size
                 if size_to_hash > PKG_FROM_FOLDER_MAX_PKG_SIZE:
```

Comment:
i'm not sure why this constant is used here 🤔

Reply:
I think at some point its meaning became `MAX_BYTES_TO_HASH`.